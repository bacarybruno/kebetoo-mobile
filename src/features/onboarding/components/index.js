import React from 'react'
import { View } from 'react-native'

import { FormatedTypography, Typography } from '@app/shared/components'
import { metrics } from '@app/theme'

import styles from './styles'

const OnboardingSlide = ({ slideTitle, slideDescription }) => {
  const showTitle = metrics.screenHeight > 568
  return (
    <View style={styles.wrapper}>
      {showTitle && (
        <Typography
          bold
          text={slideTitle}
          type={Typography.types.headline1}
          color="black"
          numberOfLines={2}
          adjustsFontSizeToFit
        />
      )}
      <View style={styles.description}>
        <FormatedTypography
          text={slideDescription}
          type={Typography.types.subheading}
          color="black"
        />
      </View>
    </View>
  )
}

export default React.memo(OnboardingSlide)
