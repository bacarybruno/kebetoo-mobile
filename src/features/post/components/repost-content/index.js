import { memo } from 'react'
import { View, TouchableWithoutFeedback } from 'react-native'

import BasicPost, { Content, POST_TYPES, getPostType } from '@app/features/post/containers/basic-post'
import { deleteProperty } from '@app/shared/helpers/object'
import { useAppStyles } from '@app/shared/hooks'

import createThemedStyles from './styles'

const RepostContent = ({
  post, originalAuthor, style, onPress, mode,
}) => {
  const styles = useAppStyles(createThemedStyles)
  return (
    <View style={[styles.wrapper, style]}>
      <TouchableWithoutFeedback onPress={onPress}>
        <View>
          <Content post={deleteProperty(post, 'repost')} type={POST_TYPES.REPOST} mode={mode} />
          <View style={[
            styles.content,
            getPostType(post.repost) === POST_TYPES.IMAGE && styles.imageContent,
          ]}
          >
            <BasicPost
              isRepost
              post={post.repost}
              withReactions={false}
              author={originalAuthor}
              mode={mode}
            />
          </View>
        </View>
      </TouchableWithoutFeedback>
    </View>
  )
}

export default memo(RepostContent)
