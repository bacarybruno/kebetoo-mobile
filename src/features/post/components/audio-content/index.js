import { memo } from 'react';
import { View } from 'react-native';

import { AudioPlayer, Typography } from '@app/shared/components';
import { extractMetadataFromName } from '@app/shared/hooks/audio-recorder';
import { env } from '@app/config';
import { useAppStyles } from '@app/shared/hooks';

import createThemedStyles from './styles';

export const getSource = (url) => (
  url.startsWith('http')
    ? url
    : `${env.assetsBaseUrl}/${url.startsWith('/') ? url.substr(1) : url}`
);

const AudioContent = ({
  content, audioName, audioUrl, style, onPress, mode,
}) => {
  const styles = useAppStyles(createThemedStyles);
  return (
    <View style={style}>
      <Typography type={Typography.types.body} text={content} style={styles.text} />
      <AudioPlayer
        onPress={onPress}
        source={getSource(audioUrl)}
        style={mode === 'comments' ? styles.comment : undefined}
        duration={parseInt(extractMetadataFromName(audioName).duration, 10)}
      />
    </View>
  );
};

export default memo(AudioContent);
