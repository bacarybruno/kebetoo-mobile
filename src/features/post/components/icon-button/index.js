import React, { useEffect, useState } from 'react'
import { TouchableOpacity, Animated, View } from 'react-native'

import Kebeticon from '@app/shared/icons/kebeticons'
import { edgeInsets, metrics } from '@app/theme'
import { Badge } from '@app/shared/components'
import { useAppColors, useAppStyles } from '@app/shared/hooks'

import createThemedStyles, { getDimensions, getIconSize } from './styles'

const scale = (value = 1) => ({ transform: [{ scale: value }] })

const IconButton = ({
  onPress,
  name,
  style,
  text = '',
  size = 40,
  activable = false,
  isActive = false,
  color,
  activeColor,
  defaultBgColor,
  defaultHitSlop = 20,
  ...otherProps
}) => {
  const [animatedScale] = useState(new Animated.Value(1))
  const styles = useAppStyles(createThemedStyles)
  const colors = useAppColors()

  useEffect(() => {
    if (activable && isActive) {
      Animated.spring(animatedScale, {
        toValue: 1.8,
        bounciness: 20,
        speed: 20,
        useNativeDriver: true,
      }).start()
    } else {
      animatedScale.setValue(1)
    }
  }, [activable, isActive, animatedScale])

  return (
    <View style={styles.wrapper}>
      {!!text && isActive && (
        <Badge style={styles.badge} text={text} />
      )}
      <TouchableOpacity
        onPress={onPress}
        activeOpacity={activable ? 1 : undefined}
        hitSlop={
          isActive
            ? edgeInsets.symmetric({
              vertical: metrics.screenHeight,
              horizontal: metrics.screenWidth,
            })
            : edgeInsets.all(defaultHitSlop)
        }
        style={[
          styles.button,
          { backgroundColor: defaultBgColor || colors.backgroundSecondary },
          isActive && styles.activeButton,
          getDimensions(size),
          scale(animatedScale),
          style,
        ]}
        {...otherProps}
      >
        <Kebeticon
          name={name}
          size={getIconSize(size)}
          color={isActive ? (activeColor || colors.white) : (color || colors.blue_dark)}
        />
      </TouchableOpacity>
    </View>
  )
}

export default React.memo(IconButton)
