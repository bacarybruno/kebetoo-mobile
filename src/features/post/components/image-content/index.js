import { memo } from 'react'
import { View, TouchableOpacity, ImageBackground } from 'react-native'
import Ionicon from 'react-native-vector-icons/Ionicons'

import { Pressable, Typography, FormatedTypography } from '@app/shared/components'
import { edgeInsets } from '@app/theme'
import { env } from '@app/config'
import { useAppColors, useAppStyles } from '@app/shared/hooks'

import createThemeColors from './styles'

export const getSource = (url) => ({ uri: url.startsWith('http') ? url : `${env.assetsBaseUrl}/${url.startsWith('/') ? url.substr(1) : url}` })

export const DeleteIconButton = ({ onPress }) => {
  const styles = useAppStyles(createThemeColors)
  const { colors } = useAppColors()
  return (
    <TouchableOpacity
      style={styles.deleteWrapper}
      onPress={onPress}
      hitSlop={edgeInsets.all(20)}
    >
      <Ionicon name="ios-close" size={25} color={colors.textPrimary} />
    </TouchableOpacity>
  )
}

export const ImageViewer = ({
  source, style, onDelete, onPress, borderRadius = 0, ...otherProps
}) => {
  const styles = useAppStyles(createThemeColors)
  return (
    <>
      <View style={[styles.imageWrapper, style, { borderRadius }]} {...otherProps}>
        <ImageBackground source={source} style={styles.flex} borderRadius={borderRadius}>
          {onPress && <Pressable style={styles.flex} onPress={onPress} />}
        </ImageBackground>
      </View>
      {onDelete && <DeleteIconButton onPress={onDelete} />}
    </>
  )
}

const ImageContent = ({
  content, url, style, mode, onPress,
}) => {
  const styles = useAppStyles(createThemeColors)
  return (
    <View style={[styles.wrapper, style, mode === 'comments' && styles.commentMode]}>
      <View style={styles.text}>
        <FormatedTypography
          type={Typography.types.body}
          text={content.trim()}
          numberOfLines={2}
        />
      </View>
      <ImageViewer
        onPress={onPress}
        style={styles.imageViewer}
        source={getSource(url)}
        testID="image-viewer"
      />
    </View>
  )
}

export default memo(ImageContent)
