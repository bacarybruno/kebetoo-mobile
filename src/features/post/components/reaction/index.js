import React from 'react'
import { TouchableOpacity } from 'react-native'

import Kebeticon from '@app/shared/icons/kebeticons'
import { edgeInsets } from '@app/theme'
import { Typography } from '@app/shared/components'
import { useAppColors } from '@app/shared/hooks'
import { abbreviateNumber } from '@app/shared/helpers/strings'

import styles from './styles'

const Reaction = ({
  iconName, count, onPress, color = 'reactions', ...otherProps
}) => {
  const { colors } = useAppColors()
  return (
    <TouchableOpacity
      style={styles.reaction}
      onPress={onPress}
      hitSlop={edgeInsets.symmetric({ horizontal: 5, vertical: 10 })}
      {...otherProps}
    >
      <Kebeticon color={colors[color]} style={styles.icon} size={18} name={iconName} />
      <Typography
        type={Typography.types.headline6}
        systemWeight={Typography.weights.bold}
        text={abbreviateNumber(count)}
        color={color}
      />
    </TouchableOpacity>
  )
}

export default Reaction
