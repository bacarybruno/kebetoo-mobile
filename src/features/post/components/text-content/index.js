import { memo } from 'react';

import { Pressable, Typography, FormatedTypography } from '@app/shared/components';
import { POST_TYPES } from '@app/features/post/containers/basic-post/helpers';
import { useAppStyles } from '@app/shared/hooks';

import createThemedStyles from './styles';

const TextContent = ({
  content, style, onPress, type, mode, isRepost,
}) => {
  const styles = useAppStyles(createThemedStyles);

  return (
    <Pressable
      onPress={onPress}
      disabled={!onPress}
      style={[
        styles.wrapper,
        type === POST_TYPES.REPOST && styles.repost,
        style,
        mode === 'comments' && styles.comments,
        mode === 'comments' && isRepost && styles.commentRepost,
      ]}
    >
      <FormatedTypography text={content.trim()} type={Typography.types.body} />
    </Pressable>
  );
};

export default memo(TextContent);
