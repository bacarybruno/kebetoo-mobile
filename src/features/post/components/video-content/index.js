import { memo } from 'react';
import { View } from 'react-native';

import { VideoPlayer, Typography, Pressable } from '@app/shared/components';
import { extractMetadataFromName } from '@app/shared/hooks/audio-recorder';
import { env } from '@app/config';
import { useAppStyles } from '@app/shared/hooks';
import { getExtension, getFileName } from '@app/shared/helpers/file';

import createThemedStyles from './styles';

export const getSource = (url) => (
  url.startsWith('http')
    ? url
    : `${env.assetsBaseUrl}/${url.startsWith('/') ? url.substr(1) : url}`
);

export const getVideoThumbnail = (videoUrl, extension = 'png') => {
  const fileName = getFileName(videoUrl);
  const fileExtension = getExtension(videoUrl);
  return videoUrl
    .replace(fileName, `thumbnails/${fileName}`)
    .replace(`.${fileExtension}`, `.${extension}`);
};

const VideoContent = ({
  content, videoName, videoUrl, localFileUri, style, onPress,
}) => {
  const styles = useAppStyles(createThemedStyles);
  return (
    <View style={style}>
      <Typography type={Typography.types.body} text={content} style={styles.text} />
      <Pressable onPress={onPress}>
        <VideoPlayer
          source={getSource(videoUrl)}
          duration={parseInt(extractMetadataFromName(videoName).duration, 10)}
          thumbnail={getVideoThumbnail(videoUrl, 'png')}
          preview={getVideoThumbnail(videoUrl, 'gif')}
          localSource={localFileUri}
        />
      </Pressable>
    </View>
  );
};

export default memo(VideoContent);
