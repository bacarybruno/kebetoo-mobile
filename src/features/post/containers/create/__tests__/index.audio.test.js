import setupTest from '@app/config/jest-setup'
import { AudioPlayer } from '@app/shared/components/audio-player'

import CreatePost, { actionTypes } from '../index'

jest.mock('@app/shared/hooks', () => ({
  useAudioRecorder: jest.fn().mockReturnValue({ hasRecording: true }),
  useAnalytics: jest.fn().mockReturnValue({ reportError: jest.fn() }),
  useAppColors: jest.fn().mockReturnValue({ colors: {} }),
  useAppStyles: jest.fn().mockReturnValue({ icon: {} }),
  useKeyboard: jest.fn().mockReturnValue({}),
  useUser: jest.fn().mockReturnValue({ profile: {} }),
}))

const givenCreatePost = setupTest(CreatePost)({
  navigation: {
    navigate: jest.fn(),
    setOptions: jest.fn().mockImplementation((params) => {
      if (params.headerRight) params.headerRight()
    }),
    dispatch: jest.fn(),
    addListener: jest.fn(),
  },
  route: {
    params: {
      action: actionTypes.SHARE,
    },
  },
})

it('renders AudioPlayer if post has recording', () => {
  const { wrapper } = givenCreatePost()
  expect(wrapper.toJSON()).toMatchSnapshot()
  expect(wrapper.root.findAllByType(AudioPlayer).length).toBe(1)
})
