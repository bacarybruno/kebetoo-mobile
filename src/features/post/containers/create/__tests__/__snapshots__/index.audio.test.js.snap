// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AudioPlayer if post has recording 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "zIndex": 1,
        },
        false,
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={null}
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderBottomColor": "rgb(216, 216, 216)",
              "flex": 1,
              "shadowColor": "rgb(216, 216, 216)",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.85,
              "shadowRadius": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "maxHeight": undefined,
            "minHeight": undefined,
            "opacity": undefined,
            "transform": undefined,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityRole="header"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 17,
                  "fontWeight": "600",
                }
              }
            >
              jest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 60,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 8,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={false}
                          hitSlop={
                            Object {
                              "bottom": 10,
                              "left": 10,
                              "right": 10,
                              "top": 10,
                            }
                          }
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                  "marginLeft": 16,
                                  "width": 24,
                                },
                                Object {
                                  "marginLeft": 0,
                                  "marginRight": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Object {
                                  "color": undefined,
                                  "fontFamily": "icomoon",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                }
                              }
                              text="kebeticon-chevron-left"
                            />
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontSize": 22,
                                    "fontWeight": "600",
                                    "letterSpacing": 0.34375,
                                    "lineHeight": 28,
                                  },
                                  Array [
                                    Object {
                                      "color": undefined,
                                    },
                                    undefined,
                                    Array [
                                      Object {
                                        "flex": 1,
                                      },
                                      undefined,
                                    ],
                                  ],
                                ]
                              }
                            >
                              Share post
                            </Text>
                            <ActivityIndicator
                              animating={false}
                              color="#999999"
                              hidesWhenStopped={true}
                              size="small"
                              style={
                                Object {
                                  "marginLeft": 8,
                                }
                              }
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      accessible={true}
                      focusable={false}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          undefined,
                          undefined,
                          Object {
                            "borderColor": undefined,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#FFFFFF",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": "600",
                              "letterSpacing": -0.32,
                              "lineHeight": 21,
                            },
                            Array [
                              Object {
                                "color": undefined,
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        SHARE
                      </Text>
                    </View>
                  </View>
                  <RCTScrollView
                    keyboardShouldPersistTaps="always"
                  >
                    <View>
                      <View
                        style={
                          Array [
                            undefined,
                            undefined,
                          ]
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontSize": 12,
                                "fontWeight": "400",
                                "letterSpacing": 0,
                                "lineHeight": 16,
                              },
                              Array [
                                Object {
                                  "color": undefined,
                                },
                                undefined,
                                undefined,
                              ],
                            ]
                          }
                        >
                          Your message
                        </Text>
                        <View
                          style={Object {}}
                        >
                          <View
                            style={
                              Array [
                                undefined,
                                undefined,
                              ]
                            }
                          >
                            <TextInput
                              allowFontScaling={true}
                              autoFocus={true}
                              blurOnSubmit={false}
                              defaultValue=""
                              editable={true}
                              maxLength={254}
                              multiline={true}
                              numberOfLines={8}
                              onBlur={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              placeholder="What's on your mind?"
                              rejectResponderTermination={true}
                              returnKeyType="default"
                              style={
                                Array [
                                  undefined,
                                  Array [
                                    undefined,
                                    undefined,
                                  ],
                                  undefined,
                                ]
                              }
                              underlineColorAndroid="transparent"
                              value=""
                            />
                          </View>
                        </View>
                        <View>
                          <View
                            style={
                              Array [
                                undefined,
                                undefined,
                              ]
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  undefined,
                                  Object {},
                                  undefined,
                                ]
                              }
                              testID="player-wrapper"
                            >
                              <View>
                                <View
                                  style={
                                    Object {
                                      "width": "0%",
                                    }
                                  }
                                  testID="progress"
                                />
                                <View
                                  accessible={true}
                                  focusable={true}
                                  hitSlop={
                                    Object {
                                      "bottom": 10,
                                      "left": 10,
                                      "right": 10,
                                      "top": 10,
                                    }
                                  }
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Object {
                                        "color": undefined,
                                        "fontFamily": "Ionicons",
                                        "fontSize": 22,
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                        "marginLeft": 3,
                                      }
                                    }
                                    text="ionicon-ios-play"
                                  />
                                </View>
                                <View>
                                  <Image
                                    source={
                                      Object {
                                        "testUri": "../../../assets/images/waves.png",
                                      }
                                    }
                                  />
                                </View>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 12,
                                        "fontWeight": "400",
                                        "letterSpacing": 0,
                                        "lineHeight": 16,
                                      },
                                      Array [
                                        Object {
                                          "color": undefined,
                                        },
                                        undefined,
                                        Array [
                                          undefined,
                                          undefined,
                                        ],
                                      ],
                                    ]
                                  }
                                >
                                  0:00
                                </Text>
                              </View>
                            </View>
                            <View
                              accessible={true}
                              focusable={true}
                              hitSlop={
                                Object {
                                  "bottom": 50,
                                  "left": 50,
                                  "right": 50,
                                  "top": 50,
                                }
                              }
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Object {
                                    "color": undefined,
                                    "fontFamily": "Ionicons",
                                    "fontSize": 20,
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  }
                                }
                                text="ionicon-ios-close"
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </RCTScrollView>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
